test/test_buffer_protocol.py

```diff
@@ -0,0 +1,163 @@
+import torch.testing._internal.common_utils as common
+from torch.testing._internal.common_device_type import (
+    instantiate_device_type_tests,
+    dtypes, onlyCPU
+)
+from torch.testing import all_types
+
+import torch
+import array
+
+TORCH_TO_ARRAYTYPE = {
+    torch.int8    : 'b',
+    torch.uint8   : 'B',
+    torch.int16   : 'h',
+    torch.int32   : 'i',
+    torch.int64   : 'q',
+    torch.float32 : 'f',
+    torch.float64 : 'd',
+}
+
+def get_dtype_size(dtype):
```

Comment:
nit: if you have a Tensor, you can use [element_size()](https://pytorch.org/docs/stable/tensors.html?highlight=element_size#torch.Tensor.element_size) method on Tensor to get that value.
But if you only have the dtype, this is good.